### PR TITLE
[MOB-5028] removes promise processing

### PR DESCRIPTION
**Pull request by evangreer91**
## 🔹 JIRA Ticket(s) if any

* [MOB-5028](https://iterable.atlassian.net/browse/MOB-5028)

## ✏️ Description

This pull request includes removing the promise processing from the SDK. The promise is returned by the SDK and processed on the application side.

### CODE REVIEW

**Review by pauljung14 — APPROVED** [review]
Lgtm, but is there a PR to update documentation? Sounds like customers would need to be aware of this in next release

File: ts/IterableInAppManager.ts [hunk 1]
```diff
@@ -102,9 +102,6 @@ class IterableInAppManager {
      Iterable.logger.log("InAppManager.getHtmlContentForMessage")
 
     return RNIterableAPI.getHtmlInAppContentForMessage(message.messageId)
-      .then((content: any) => {
-        return IterableHtmlInAppContent.fromDict(content)
-      })
```
Comment (Ayyanchira) [inline ts/IterableInAppManager.ts:107]: Wow..  the `.then` part was unnecessary the whole time? 
Have you verified if its properly returning just the HTML part?

**Review by Ayyanchira — APPROVED** [review]
👌🏼 Wow.. thats such a great find Evan!

File: ts/IterableInAppManager.ts (hunk 1 above)
Comment (evangreer91) [reply ts/IterableInAppManager.ts:107]: Yes, but I will verify before merging.

**Comment by evangreer91** [comment]
> Lgtm, but is there a PR to update documentation? Sounds like customers would need to be aware of this in next release

This does not change what is returned by the APIs. So, implementation on the application side is unchanged.

**Pull request merged**